libpod/oci_util.go

```diff
@@ -83,11 +83,11 @@ func bindPorts(ports []ocicni.PortMapping) ([]*os.File, error) {
 
 func getOCIRuntimeError(runtimeMsg string) error {
 	r := strings.ToLower(runtimeMsg)
-	if match, _ := regexp.MatchString(".*permission denied.*|.*operation not permitted.*", r); match {
-		return errors.Wrapf(define.ErrOCIRuntimePermissionDenied, "%s", strings.Trim(runtimeMsg, "\n"))
+	if match := regexp.MustCompile(".*permission denied.*|.*operation not permitted.*").FindString(r); match != "" {
+		return errors.Wrapf(define.ErrOCIRuntimePermissionDenied, "%s", strings.Trim(match, "\n"))
 	}
-	if match, _ := regexp.MatchString(".*executable file not found in.*|.*no such file or directory.*", r); match {
-		return errors.Wrapf(define.ErrOCIRuntimeNotFound, "%s", strings.Trim(runtimeMsg, "\n"))
+	if match := regexp.MustCompile(".*executable file not found in.*|.*no such file or directory.*").FindString(r); match != "" {
+		return errors.Wrapf(define.ErrOCIRuntimeNotFound, "%s", strings.Trim(match, "\n"))
```

Comment:
shouldn't we still have the runtimeMsg value in the error message?  I'd suggest:

```
 		return errors.Wrapf(define.ErrOCIRuntimeNotFound, "runtimeMsg %s returned a match of %s", runtimeMsg, strings.Trim(match, "\n")) 
```
Ditto line 87

Reply:
the idea of the PR is to not include `runtimeMsg` because it could contain other output

Reply:
Personally I think it would be easier to have it in the error message so you can try and figure out what happened.  My $.02.

Reply:
I've changed it so that the full output is still printed when `--log-level=debug` is used